.github/workflows/python-app.yml

```diff
@@ -0,0 +1,38 @@
+# This workflow will install Python dependencies, and run tests from the Odemis repository.
+# It uses the python 3 system interpreter to run test cases.
+
+name: Unit testing
+
+on:
+  [push]
+jobs:
+  build:
+
+    runs-on: ubuntu-18.04
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install Odemis dependencies
+      run: |
+        sudo add-apt-repository ppa:delmic-soft/odemis
+        sudo apt-get -y update
+
+        sudo apt-get -y install python3-pil python3-yaml python3-serial python3-zmq \
+        python3-scipy python3-decorator cgroup-bin \
+        python3-libtiff python3-comedilib python3-h5py libtiff5-dev python3-future \
+        python3-pkg-resources python3-netifaces python3-cairo python3-numpy python3-scipy \
+        python3-decorator python3-xdg python3-suds python3-matplotlib python3-opencv \
+        python3-wxgtk4.0 python3-serial python3-evdev python3-notify2 python3-psutil \
+        graphicsmagick cgroup-bin authbind python3-setuptools cython python3-pyro4-delmic
+
+        sudo mkdir /var/run/odemisd
+        sudo chmod a+rw /var/run/odemisd
+
+    - name: Run tests from odemis.model
+      run: |
+        export PYTHONPATH="/home/runner/work/odemis/odemis/src:$PYTHONPATH"
+        python3 -m unittest src/odemis/model/test/components_test.py
+        python3 -m unittest src/odemis/model/test/dataflow_test.py
+        python3 -m unittest src/odemis/model/test/futures_test.py
```

Comment:
@pieleric I just noticed that when one of these lines has a failure it doesn't execute the next line. There was a failure in futures_test, and it didn't execute remote_test and vattributes_test. Should I put it in separate run blocks? If we have to put it in separate run blocks the file will eventually become quite chaotic and long.

Reply:
Yes, I think that's the right approach.
Possibly, in the future we'd have a special version of the util/runtest.sh script to run in a pipeline (just the tests which are possible).